compiler/optimizer/VPHandlers.cpp

```diff
@@ -1476,8 +1513,13 @@ TR::Node *constrainAload(TR::ValuePropagation *vp, TR::Node *node)
       return node;
 
    // before undertaking aload specific handling see if the default load transformation helps
-   if (TR::TransformUtil::transformDirectLoad(vp->comp(), node))
+   // But even if it does help, node may still be an aload that we can constrain.
+   if (TR::TransformUtil::transformDirectLoad(vp->comp(), node)
+       && node->getOpCodeValue() != TR::aload)
```

Comment:
I feel it's preferable to ensure that, when continuing on within `constrainAload`, the node is in fact an `aload`. It should be safe to pass any node into `constrainNewlyFoldedConst`, which will gracefully ignore opcodes that it doesn't recognize.